docs/blog/pydantic-v2.md

```diff
@@ -0,0 +1,743 @@
+
+### No pure python implementation :frowning:
+
+Since pydantic-core is written in Rust, and I have absolutely no intention of rewriting it in python,
+pydantic V2 will only work where a binary package can be installed.
+
+pydantic-core will provide binaries in PyPI for (at least):
+
+* **Linux**: `x86_64`, `aarch64`, `i686`, `armv7l`, `musl-x86_64` & `musl-aarch64`
+* **MacOS**: `x86_64` & `arm64` (except python 3.7)
+* **Windows**: `amd64` & `win32`
+* **Web Assembly**: `wasm32` 
+  (pydantic-core is [already](https://github.com/samuelcolvin/pydantic-core/runs/7214195252?check_suite_focus=true) 
+  compiled for wasm32 using emscripten and unit tests pass, except where cpython itself has 
+  [problems](https://github.com/pyodide/pyodide/issues/2841))
+
+Other binaries can be added provided they can be (cross-)compiled on github actions.
+If no binary is available from PyPI, pydantic-core can be compiled from source if Rust stable is available.
+
+The only place where I know this will cause problems is Raspberry Pi, which is a 
+[mess](https://github.com/piwheels/packages/issues/254) when it comes to packages written in Rust for python.
+Effectively, until that's fixed you'll likely have to install pydantic with 
+`pip install -i https://pypi.org/simple/ pydantic`.
+
+### Pydantic becomes a pure python package :thumbsup:
+
+Pydantic V1.X is a pure python code base but is compiled with cython to provide some performance improvements.
+Since the "hot" code is moved to pydantic-core, pydantic itself can go back to being a pure python package.
+
+This should significantly reduce the size of the pydantic package and make unit tests of pydantic much faster.
+In addition, some constraints on pydantic code can be removed once it no-longer has to be compilable with cython.
+
+Some pieces of edge logic could get a little slower as they're no longer compiled.
+
+### I'm dropping the word "parse" and just using "validate" :neutral_face:
+
+Partly due to the issues with the lack of strict mode,
+I've gone back and forth between using the terms "parse" and "validate" for what pydantic does.
+
+While pydantic is not simply a validation library (and I'm sure some would argue validation is not strictly what it does),
+most people use the word **"validation"**.
+
+It's time to stop fighting that, and use consistent names.
+
+The word "parse" will no longer be used except when talking about JSON parsing, see
+[model methods](#model-namespace-cleanup) above.
+
+## Changes to custom field types :neutral_face:
+
+Since the core structure of validators has changed from "a list of validators to call one after another" to
+"a tree of validators which call each other", the 
+[`__get_validators__`](https://pydantic-docs.helpmanual.io/usage/types/#classes-with-__get_validators__)
+way of defining custom field types no longer makes sense.
+
+Instead we'll look for the attribute `__pydantic_schema__` which must be a 
```

Comment:
This all makes sense, the only thing I wonder about is the word "**schema**", because it's the same word used for JSON Schema, and I'm not sure if that could make some confusion.

Also because both types of "schemas" are mainly a dict with string keys.

Reply:
Since it's a pydantic-core schema and that's unlikely to change, maybe `__pydantic_core_schema__`?

Reply:
Humm, dang it, "naming things", that's an NP-Complete problem :joy: 

As @adriangb says, maybe something that explicitly mentions `core` or `validation` could work and help disambiguate.

Maybe it's not that important, but it's just that I have "suffered" and regretted design choices I've made in the past, e.g. naming two different things with the same name "`app`", and then having to deal with disambiguating that everywhere, and then having to keep the two ambiguous names for backwards compatibility and all that.

Anyway, some word ideas, probably none is perfect, but in case you like any of those:

* definition
* instruction
* declaration
* shape
* design
* architecture
* plan
* pattern
* blueprint
* structure
* arrangement
* prescription
* skeleton
* outline
* frame
* formula

Reply:
We could also rename the `schema` method into `model_json_schema` instead of `model_schema`.
I know there is also the `model_schema_json` but IMHO it could be removed in v2. Don't really see the advantage compared to a simple `(or)json.dumps...` of the json schema as dict

Reply:
Okay, I've called this thing `__pydantic_validation_schema__`.

@PrettyWood @tiangolo if you can be bothered, it might be worth you looking at the new table I added in "Model Namespace Cleanup" section.